apps/explorer/src/ui/CopyToClipboard.tsx

```diff
@@ -28,31 +34,65 @@ const iconStyles = cva(['cursor-pointer hover:text-steel-dark'], {
 
 export type IconStylesProps = VariantProps<typeof iconStyles>;
 
-export interface CopyToClipboardProps extends IconStylesProps {
+export interface CopyToClipboardProps extends Omit<IconStylesProps, 'success'> {
 	copyText: string;
 	onSuccessMessage?: string;
 }
 
-const ICON_SIZES = {
+const COPY_ICON_SIZES = {
 	sm: Copy12,
 	md: Copy16,
 	lg: Copy24,
 };
 
+const CHECK_ICON_SIZES = {
+	sm: Check12,
+	md: CheckStroke16,
+	lg: CheckStroke24,
+};
+
+const TIMEOUT_TIMER = 2000;
+
 export function CopyToClipboard({
 	copyText,
 	color,
 	size = 'md',
 	onSuccessMessage = 'Copied!',
 }: CopyToClipboardProps) {
+	const [showCopyIcon, setShowCopyIcon] = useState(true);
 	const copyToClipBoard = useCopyToClipboard(() => toast.success(onSuccessMessage));
 
-	const CopyIcon = ICON_SIZES[size!];
+	const CopyIcon = COPY_ICON_SIZES[size!];
+	const CheckIcon = CHECK_ICON_SIZES[size!];
+
+	const onClickHandler = async () => {
+		await copyToClipBoard(copyText);
+		setShowCopyIcon(false);
+	};
+
+	useEffect(() => {
+		if (!showCopyIcon) {
+			const timeout = setTimeout(() => {
+				setShowCopyIcon(true);
+			}, TIMEOUT_TIMER);
+			return () => clearTimeout(timeout);
+		}
+	}, [showCopyIcon]);
 
 	return (
-		<Link onClick={() => copyToClipBoard(copyText)}>
+		<Link disabled={!showCopyIcon} onClick={onClickHandler}>
 			<span className="sr-only">Copy</span>
-			<CopyIcon className={iconStyles({ size, color })} />
+			{showCopyIcon ? (
+				<motion.div
```

Comment:
is framer-motion overkill for this animation since it's just transitioning the opacity?

Reply:
I tried but it's a bit more complicated bc tailwind works if u do something like `opacity-100 hover:opacity-0 transition-opacity duration-1000 ease-in` but when it needs to be triggered on mount every time to transition from opacity-0 to opacity-100, it does not run correctly since both the opacity values need to be presented. I could append/remove the className but that would be more complicated than just use framer motion.

Reply:
I'd actually vote to remove this altogether unless it was a specific request from @mystie711 ... I don't think we're gaining anything with the transition here

Reply:
With a second look at this...let's increase the transition time to 200ms.